plugin.py

```diff
@@ -136,7 +138,37 @@ def on_pre_start(
         configuration.command = cls.get_command()
         return None
 
-    # notification handlers
+    # -- commands from the server that should be handled client-side ----------
+
+    def on_pre_server_command(
+        self,
+        command: Mapping[str, Any],
+        done_callback: Callable[[], None]
+    ) -> bool:
+        name = command["command"]
+        if name == "omnisharp/client/findReferences":
+            return self._handle_quick_references(command["arguments"], done_callback)
+        return False
+
+    def _handle_quick_references(self, arguments: List[Any], done_callback: Callable[[], None]) -> bool:
+        session = self.weaksession()
+        if not session:
+            return False
+        sb = session.get_session_buffer_for_uri_async(arguments[0]["uri"])
+        if not sb:
+            return False
+        for sv in sb.session_views:
+            if not sv.view.is_valid():
+                continue
+            r = Range.from_lsp(arguments[0]["range"])
+            region = range_to_region(r, sv.view)
+            args = {"point": region.a}
+            sv.view.run_command("lsp_symbol_references", args)
+            done_callback()
+            return True
+        return False
```

Comment:
Do we actually want to return False at all? Will server handle that command correctly if client can't do it?